test/parallel/test-require-deps-deprecation.js

```diff
@@ -39,6 +39,16 @@ for (const m of deprecatedModules) {
   } catch (err) {}
 }
 
+// Instead of checking require, check that resolve isn't pointing toward
+// built-in module, as user might already have node installed with acorn in
+// require.resolve range
+// Ref: https://github.com/nodejs/node/issues/17148
 for (const m of deps) {
-  assert.throws(() => { require(m); }, /^Error: Cannot find module/);
+  let path;
+  try {
+    path = require.resolve(m);
+  } catch (err) {
+    assert.ok(err.toString().startsWith('Error: Cannot find module '));
```

Comment:
I’d still like to see a `continue` here to avoid unneeded assertion in the following line.